server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -46,33 +55,27 @@
 import org.elasticsearch.search.profile.query.QueryTimingType;
 
 import java.io.IOException;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Set;
 
 /**
  * Context-aware extension of {@link IndexSearcher}.
  */
 public class ContextIndexSearcher extends IndexSearcher implements Releasable {
-
-    /** The wrapped {@link IndexSearcher}. The reason why we sometimes prefer delegating to this searcher instead of {@code super} is that
-     *  this instance may have more assertions, for example if it comes from MockInternalEngine which wraps the IndexSearcher into an
-     *  AssertingIndexSearcher. */
-    private final XIndexSearcher in;
+    private static int CHECK_CANCELLED_INTERVAL = 1 << 11;
 
     private AggregatedDfs aggregatedDfs;
-
-    private final Engine.Searcher engineSearcher;
-
-    // TODO revisit moving the profiler to inheritance or wrapping model in the future
     private QueryProfiler profiler;
-
     private Runnable checkCancelled;
 
-    public ContextIndexSearcher(Engine.Searcher searcher, QueryCache queryCache, QueryCachingPolicy queryCachingPolicy) {
-        super(searcher.reader());
-        engineSearcher = searcher;
-        in = new XIndexSearcher(searcher.searcher());
-        setSimilarity(searcher.searcher().getSimilarity());
+    public ContextIndexSearcher(IndexReader reader) {
```

Comment:
I wonder if we should really expose this? it seems only used in tests, maybe we should enforce passing all the other stuff.

Reply:
agreed, this is trappy. I removed it: https://github.com/elastic/elasticsearch/pull/43645/commits/45baded6ece7ca2c978a84faad0dec2c981c8a5a